pkg/engine/deployment.go

```diff
@@ -183,6 +185,9 @@ func newDeployment(ctx *Context, info *deploymentContext, opts *deploymentOption
 		return nil, err
 	}
 
+	if plugctx.Host != nil && opts.Loader == nil {
```

Comment:
Shouldn't Host always be non-nil by this point?

Reply:
That I am not 100% sure about